nwss_wastewater/delphi_nwss/pull.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""Functions for pulling NCHS mortality data API."""
+
+import numpy as np
+import pandas as pd
+from sodapy import Socrata
+
+from .constants import (
+    SIGNALS,
+    METRIC_SIGNALS,
+    METRIC_DATES,
+    SAMPLE_SITE_NAMES,
+    SIG_DIGITS,
+    NEWLINE,
+)
+
+
+def sig_digit_round(value, n_digits):
+    """Round the number of significant digits (x.xxe5 would be 3) to `N`."""
+    in_value = value
+    value = np.asarray(value).copy()
+    zero_mask = (value == 0) | np.isinf(value) | np.isnan(value)
+    value[zero_mask] = 1.0
+    sign_mask = value < 0
+    value[sign_mask] *= -1
+    exponent = np.ceil(np.log10(value))
+    result = 10**exponent * np.round(value * 10 ** (-exponent), n_digits)
+    result[sign_mask] *= -1
+    result[zero_mask] = in_value[zero_mask]
+    return result
+
+
+def construct_typedicts():
+    """Create the type conversion dictionary for both dataframes."""
+    # basic type conversion
+    signals_dict = {key: float for key in SIGNALS}
+    type_dict = {**signals_dict}
+    type_dict["timestamp"] = "datetime64[ns]"
+    # metric type conversion
+    signals_dict_metric = {key: float for key in METRIC_SIGNALS}
+    metric_dates_dict = {key: "datetime64[ns]" for key in METRIC_DATES}
+    type_dict_metric = {**metric_dates_dict, **signals_dict_metric, **SAMPLE_SITE_NAMES}
+    return type_dict, type_dict_metric
+
+
+def warn_string(df, type_dict):
+    """Format the warning string."""
+    return f"""
+Expected column(s) missed, The dataset schema may
+have changed. Please investigate and amend the code.
+
+Columns needed:
+{NEWLINE.join(type_dict.keys())}
+
+Columns available:
+{NEWLINE.join(df.columns)}
+"""
+
+
+def reformat(df, df_metric):
```

Comment:
would it make sense to call this `combine()` instead of `reformat()`?

Reply:
I'll eventually be adding some stuff that adds metric somewhat separately, and that probably won't use this reformatted table, so I think `combine` doesn't really make sense. How about `add_population`?

Reply:
sounds good!